src/Symfony/Component/Validator/Constraints/WeekValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class WeekValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Week) {
+            throw new UnexpectedTypeException($constraint, Week::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if (!preg_match('/^\d{4}-W(0[1-9]|[1-4][0-9]|5[0-3])$/', $value)) {
+            $this->context->buildViolation($constraint->message)
+                ->setCode(Week::INVALID_FORMAT_ERROR)
+                ->addViolation();
+        }
+
+        if ($constraint->strict) {
```

Comment:
This should probably only be done if the condition above didn't already add a violation.

Reply:
Indeed! Updated to an elseif